common/common_structs/src/unique_payments.rs

```diff
@@ -0,0 +1,133 @@
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+use mergeable::Mergeable;
+
+pub type PaymentsVec<M> = ManagedVec<M, EsdtTokenPayment<M>>;
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, Clone, PartialEq, Debug)]
+pub struct UniquePayments<M: ManagedTypeApi> {
+    payments: PaymentsVec<M>,
+}
+
+impl<M: ManagedTypeApi> Default for UniquePayments<M> {
+    #[inline]
+    fn default() -> Self {
+        Self {
+            payments: PaymentsVec::new(),
+        }
+    }
+}
+
+impl<M: ManagedTypeApi> UniquePayments<M> {
+    #[inline]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    #[inline]
+    pub fn new_from_unique_payments(payments: PaymentsVec<M>) -> Self {
+        UniquePayments { payments }
+    }
+
+    pub fn new_from_payments(payments: PaymentsVec<M>) -> Self {
+        let mut merged_payments = Self::new();
+        for p in &payments {
+            merged_payments.add_payment(p);
+        }
+
+        merged_payments
+    }
+
+    pub fn add_payment(&mut self, new_payment: EsdtTokenPayment<M>) {
+        if new_payment.amount == 0 {
+            return;
+        }
+
+        let len = self.payments.len();
+        for i in 0..len {
+            let mut current_payment = self.payments.get(i);
+            if !current_payment.can_merge_with(&new_payment) {
+                continue;
+            }
+
+            current_payment.amount += new_payment.amount;
+            let _ = self.payments.set(i, &current_payment);
+
+            return;
+        }
+
+        self.payments.push(new_payment);
+    }
+    #[allow(clippy::result_unit_err)]
```

Comment:
Is the return type of the function necessary here? Or is there any way to avoid the clippy tag?